dissect/target/plugins/os/unix/log/packagemanagers/zypper.py

```diff
@@ -0,0 +1,111 @@
+
+        self.target_timezone = target.datetime.tzinfo
+
+    def check_compatible(self):
+        if not self.target.fs.path(self.LOGS_DIR_PATH).glob(self.LOGS_GLOB):
+            raise UnsupportedPluginError("No zypper logs found")
+
+    @staticmethod
+    def parse_logs(log_lines: [str], tz: ZoneInfo) -> Iterator[PackageManagerLogRecord]:
+        """
+        Logs are formatted like this:
+            2022-12-16 12:56:23|command|root@ec9fa6d67dda|'zypper' 'install' 'unzip'|
+            2022-12-16 12:56:23|install|update-alternatives|1.21.8-1.4|x86_64||repo-oss|b4d6389437e306d6104559c82d09fce15c4486fbc7fd215cc33d265ff729aaf1|  # noqa
+            # 2022-12-16 12:56:23 unzip-6.00-41.1.x86_64.rpm installed ok
+            # Additional rpm output:
+            # update-alternatives: using /usr/bin/unzip-plain to provide /usr/bin/unzip (unzip) in auto mode
+            #
+            2022-12-16 12:56:23|install|unzip|6.00-41.1|x86_64|root@ec9fa6d67dda|repo-oss|d7e42c9d83f97cf3b7eceb4d3fa64e445a33a7a33f387366734c444d5571cb3a|  # noqa
+            2022-12-16 12:57:50|command|root@ec9fa6d67dda|'zypper' 'remove' 'unzip'|
+            # 2022-12-16 12:57:50 unzip-6.00-41.1.x86_64 removed ok
+            # Additional rpm output:
+            # update-alternatives: warning: alternative /usr/bin/unzipsfx-plain (part of link group unzipsfx) doesn't exist; removing from list of alternatives  # noqa
+            # update-alternatives: warning: alternative /usr/bin/zipgrep-plain (part of link group zipgrep) doesn't exist; removing from list of alternatives  # noqa
+            #
+            2022-12-16 12:57:50|remove |unzip|6.00-41.1|x86_64|root@ec9fa6d67dda|
+            2022-12-16 12:58:49|command|root@ec9fa6d67dda|'zypper' 'install' 'unzip'|
+        """  # noqa E501
+
+        for line in log_lines:
+            # we don't parse additional output logs or empty logs
+            if line.startswith("#") or line == "":
+                continue
+
+            ts, operation, *log_arguments = line.split("|")
+            ts = datetime.strptime(ts, "%Y-%m-%d %H:%M:%S").replace(tzinfo=tz)
+            operation = OperationTypes.infer(operation)
+
+            record = PackageManagerLogRecord(package_manager="zypper", ts=ts, operation=operation.value)
+
+            if operation == OperationTypes.Install:
+                yield parse_install_line(log_arguments, record)
+            elif operation == OperationTypes.Remove:
+                yield parse_remove_line(log_arguments, record)
+            elif operation == OperationTypes.Other:
+                yield parse_command_line(log_arguments, record)
+
+    @plugin.export(record=PackageManagerLogRecord)
+    def logs(self):
+        log_file_paths = self.target.fs.path(self.LOGS_DIR_PATH).glob(self.LOGS_GLOB)
+
+        for path in log_file_paths:
+            log_lines = [line.strip() for line in open_decompress(path, "rt")]
+
+            yield from self.parse_logs(log_lines, self.target_timezone)
```

Comment:
Same comment here as the webserver PR, probably better off moving most parsing back into one method.